libs/ui/editor/src/lib/listView/use-selectable-list.tsx

```diff
@@ -0,0 +1,46 @@
+import {useEffect, useState} from 'react'
+
+type UseSelectableListProps = {
+  ids?: string[]
+}
+
+export function useSelectableList({ids}: UseSelectableListProps) {
+  const [selectedItems, setSelectedItems] = useState<string[]>([])
+  const allSelected = ids && selectedItems.length === ids.length
+  const someSelected = ids && selectedItems.length > 0
+
+  const toggleAll = () => setSelectedItems(someSelected ? [] : [...(ids ?? [])])
```

Comment:
```suggestion
  const toggleAll = () => setSelectedItems(someSelected ? [] : ids ?? []])
```

Should do the trick, no?

Reply:
Yeah, it should work. Was worried about reference, but it works fine. I will do my own change as there is 1 extra bracket in change 😎